pkg/reconciler/delivery/table_test.go

```diff
@@ -61,15 +68,87 @@ func TestReconcile(t *testing.T) {
-				WithSpecTraffic(v1.TrafficTarget{ConfigurationName: "test", Percent: ptr.Int64(100)}),
+			)},
+		},
+	}, {
+		Name: "sets a 90/10 split when R2 enters",
+		Key:  "default/test",
+		Objects: []runtime.Object{
+			Route("default", "test", WithConfigTarget("test"), WithRouteGeneration(1), WithStatusTraffic(v1.TrafficTarget{
+				LatestRevision: ptr.Bool(true),
+				Percent: ptr.Int64(100),
+				RevisionName: "R1",
+			})),
+			Configuration("default", "test", WithLatestCreated("R2"), WithLatestReady("R2")),
+		},
+		WantUpdates: []clientgotesting.UpdateActionImpl{
+			{Object: Route("default", "test",
+				WithSpecTraffic(v1.TrafficTarget{
+					RevisionName: "R1",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(90),
+				}, v1.TrafficTarget{
+					RevisionName: "R2",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(10),
+				}),
+				WithRouteGeneration(1),
+				WithStatusTraffic(v1.TrafficTarget{
+					LatestRevision: ptr.Bool(true),
+					Percent: ptr.Int64(100),
+					RevisionName: "R1",
+				}),
+				WithRouteAnnotation(map[string]string{AnnotationKey: now.Format(TimeFormat)}),
+			)},
+		},
+	}, {
+		Name: "progresses to 50/50 with timestamp expiration",
+		Key:  "default/test",
+		Objects: []runtime.Object{
+			Route("default", "test", WithConfigTarget("test"), WithRouteGeneration(2), WithStatusTraffic(v1.TrafficTarget{
+				RevisionName: "R1",
+				LatestRevision: ptr.Bool(false),
+				Percent: ptr.Int64(90),
+			}, v1.TrafficTarget{
+				RevisionName: "R2",
+				LatestRevision: ptr.Bool(false),
+				Percent: ptr.Int64(10),
+			}),
+			// we want the reconciler to think that the Route was last updated 25 seconds ago
+			WithRouteAnnotation(map[string]string{AnnotationKey: now.Add(-25 * time.Second).Format(TimeFormat)})),
+			Configuration("default", "test", WithLatestCreated("R2"), WithLatestReady("R2")),
+		},
+		WantUpdates: []clientgotesting.UpdateActionImpl{
+			{Object: Route("default", "test",
+				WithSpecTraffic(v1.TrafficTarget{
+					RevisionName: "R1",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(50),
+				}, v1.TrafficTarget{
+					RevisionName: "R2",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(50),
+				}),
+				WithRouteGeneration(2),
+				WithStatusTraffic(v1.TrafficTarget{
+					RevisionName: "R1",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(90),
+				}, v1.TrafficTarget{
+					RevisionName: "R2",
+					LatestRevision: ptr.Bool(false),
+					Percent: ptr.Int64(10),
+				}),
+				WithRouteAnnotation(map[string]string{AnnotationKey: now.Format(TimeFormat)}),
 			)},
 		},
 	}}
 	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
 		r := &Reconciler{
 			client:      servingclient.Get(ctx),
 			routeLister: listers.GetRouteLister(),
-			// followup:    
+			// it isn't possible to test event queue here, I guess... is it?
```

Comment:
nit: use a todo format. `TODO: test event queue here`

Yes, you can attach any `PostConditions`: https://github.com/googleinterns/knative-continuous-delivery/blob/b01e6028adfcdfdf667d9357430f65d8e785c878/vendor/knative.dev/pkg/reconciler/testing/table.go#L91

Reply:
I am now using TODO. I still haven't figured out how exactly the PostConditions work. I will add that part later.